docs/conf.py

```diff
@@ -5,6 +5,26 @@
 
 from pkg_resources import get_distribution
 
+
+def _add_annotations_import(path):
+    with open(path, "r+") as f:
+        contents = f.read()
+        if not contents.startswith("from __future__ import annotations"):
+            f.seek(0, 0)
+            f.write("from __future__ import annotations  " + contents)
+
+
+def _recursive_add_annotations_import():
+    for path, _, files in os.walk("../jaxsim/"):
+        for file in [f for f in files if f.endswith(".py")]:
+            _add_annotations_import(os.path.join(path, file))
+
+
+if "READTHEDOCS" in os.environ:
+    _recursive_add_annotations_import()
```

Comment:
Do you have any reference for the need of this logic?

Reply:
You may have a look at the [PEP 563](https://peps.python.org/pep-0563/) on postponed type evaluation. The aim here is to re-evaluate the types defined in the typing module so that they appear correctly in the documentation